lib/src/router.dart

```diff
@@ -30,6 +31,9 @@ GoRouter buildRouter([GlobalKey<NavigatorState>? key]) => GoRouter(
           GoRoute(
               path: SignInView.routeName,
               builder: (_, __) => const SignInView()),
+          GoRoute(
```

Comment:
Just thought about it but we can use the `GoRoute.redirect` method to check if the user already has an account and redirect him/her to the profile if he/she already has. Blocking the page to only users without an account. ([doc](https://pub.dev/documentation/go_router/latest/topics/Redirection-topic.html)).

We can also update the sign in route the same way 🤔 once the login is added

Reply:
Good point, but it needs to be in its own issue as nothing is managed in this ticket